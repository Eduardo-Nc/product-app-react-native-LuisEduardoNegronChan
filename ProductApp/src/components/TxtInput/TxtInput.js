import React from 'react'
import { Text, StyleSheet, TextInput, View } from 'react-native'

//Constants
import { SIZES, COLORS, ICONS } from '@Utils/constants';

const TxtInput = ({ field = "", value = "", title = "", type = "default", placeholder = "", onChangeTxt, isEditable = true }) => {

  return (
    <View style={styles.container}>
      <Text testID="txtTextInput" style={styles.txtTitleInputs}>{title}</Text>
      <TextInput
        editable={isEditable}
        placeholder={placeholder}
        style={styles.txtInput}
        value={value.toString()}
        keyboardType={type}
        onChangeText={onChangeTxt(field)}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
    paddingHorizontal: 20,
    marginBottom: 15
  },
  txtTitleInputs: {
    fontSize: 16,
    fontWeight: "400",
    color: COLORS.black,
    marginBottom: 5

  },
  txtInput: {
    width: "100%",
    height: 45,
    color: COLORS.black,
    borderWidth: 0.5,
    borderColor: COLORS.black,
    borderRadius: 10,
    paddingHorizontal: 15
  }

});
export default TxtInput