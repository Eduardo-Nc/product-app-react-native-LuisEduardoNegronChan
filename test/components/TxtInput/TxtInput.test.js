import { render, screen, fireEvent } from '@testing-library/react-native';
import TxtInput from '@Components/TxtInput/TxtInput';

describe('Prueba del componente TxtInput', () => {

  let testID = "txtTextInput"
  let initialText = "Titulo"
  let functionCompleted = 1
  const handleSaveClickMock = jest.fn();

  beforeEach(() => {
    render(
      <TxtInput onChangeTxt={handleSaveClickMock} title={initialText} />
    );
  })

  it('Validacion del componente con parametros iniciales', async () => {
    let result = (await screen.findByTestId(testID)).props
    expect(result.children).toBe(initialText)

    fireEvent.press(screen.getByText(initialText));
    expect(handleSaveClickMock).toHaveBeenCalledTimes(functionCompleted);
  });

  it('Muestra el placeholder', () => {
    const placeholder = "Correo electr√≥nico";
    render(<TxtInput placeholder={placeholder} onChangeTxt={jest.fn()} />);
    expect(screen.getByPlaceholderText(placeholder)).toBeTruthy();
  });

  it('Valida integridad de cambios en el componente', () => {
    expect(screen).toMatchSnapshot();
  });
});