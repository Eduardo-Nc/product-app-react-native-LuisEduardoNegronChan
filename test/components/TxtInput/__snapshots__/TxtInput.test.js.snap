// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente TxtInput Valida integridad de cambios en el componente 1`] = `
<View
  style={
    {
      "marginBottom": 15,
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "400",
        "marginBottom": 5,
      }
    }
    testID="txtTextInput"
  >
    Titulo
  </Text>
  <TextInput
    editable={true}
    keyboardType="default"
    placeholder=""
    style={
      {
        "borderColor": "#000000",
        "borderRadius": 10,
        "borderWidth": 0.5,
        "color": "#000000",
        "height": 45,
        "paddingHorizontal": 15,
        "width": "100%",
      }
    }
    value=""
  />
</View>
`;
